encode.go

```diff
@@ -484,7 +484,7 @@ func FormatTimestamp(t time.Time) []byte {
 		t = t.AddDate((-t.Year())*2+1, 0, 0)
 		bc = true
 	}
-	b := []byte(t.Format("2006-01-02 15:04:05.999999999Z07:00"))
+	b := []byte(t.Truncate(time.Microsecond).Format("2006-01-02 15:04:05.999999999Z07:00"))
```

Comment:
Some applications may prefer to round time, or round up, round down, or to truncate. Should we make this configurable? It's arguably likely to be unimportant so I'd personally be fine with truncation only, but just wondering.

Reply:
The example that we ran into was with https://github.com/jinzhu/now and using the EndOfDay() of a time. It would give the exact moment before the end of a day.

Inserting that into postgres, then reading it back out (with the rounding on insert), then the output will claim it was 00:00:00... of the next day - not the same day.

Reply:
**The current issue with over accurate data in Postgres**
*`postgres-database=# select '2021-03-21T23:59:59.9999999Z'::timestamptz;  
timestamptz*
2021-03-22 00:00:00+00`
**The proposed solution**
*`postgres-database=# select '2021-03-21T23:59:59.999999Z'::timestamptz;  
timestamptz*
2021-03-21 23:59:59.999999+00`

Reply:
I agree you're thought is correct too. If I had to guess, though, most Go developers will not truncate their times before giving it off to the drivers. In the situation you're dealing with contracts/time sensitive situations, it may make the difference of a customer/service hearing the current day versus the next day.